synapse/api/auth.py

```diff
@@ -454,7 +451,23 @@ def _verify_third_party_invite(self, event, auth_events):
                         decode_base64(public_key)
                     )
                     verify_signed_json(signed, server, verify_key)
-                    return True
+
+                    @defer.inlineCallbacks
+                    def check_key_valid():
+                        try:
+                            response = yield self.hs.get_simple_http_client().get_json(
+                                invite_event.content["key_validity_url"],
+                                {"public_key": public_key}
+                            )
+                        except Exception:
+                            raise AuthError(
+                                502,
+                                "Third party certificate could not be checked"
+                            )
```

Comment:
I suspect that this would be better as a `SynapseError`, so that we're absolutely sure we don't persist the fact that this was rejected.